app/server/api/auth_api.py

```diff
@@ -229,6 +225,23 @@ def post(self):
 
             auth_token = user.encode_auth_token()
 
+            db.session.flush()
+
+            # Possible Outcomes:
+            # TFA required, but not set up
+            # TFA not required
+
+            tfa_response_oject = tfa_logic(user, tfa_token=None)
+            if tfa_response_oject:
+                tfa_response_oject['auth_token'] = auth_token.decode()
+
+                db.session.commit()  # need to commit here so that is_activated = True
+
+                return make_response(jsonify(tfa_response_oject)), 401
+
+            # Update the last_seen TS for this user
+            user.update_last_seen_ts()
+
```

Comment:
adding tfa flow to `/activate`

Reply:
@enjeyw you're happy with this logic being added here?

see: https://github.com/teamsempo/SempoBlockchain/pull/91#discussion_r363072911